polyfills/settings/service/js/app.js

```diff
@@ -85,13 +85,25 @@
       _settings.onsettingchange = observerTemplate;
     } else {
       // It's either a get or a set... or an error but let's assume it isn't :P
-      _locks[requestOp.lockId][requestOp.operation](requestOp.settings).
-        then(result => {
-          channel.postMessage({
-            remotePortId: remotePortId,
-            data: { id : request.id, result: result}}
-          );
-      });
+      if (_locks[requestOp.lockId].closed) {
+        _locks[requestOp.lockId] = _settings.createLock();
+      }
+      var req =
+        _locks[requestOp.lockId][requestOp.operation](requestOp.settings);
+
```

Comment:
Since req is a promise now, you can write this with the then (as it was, adding a reject clause if you want to catch the error also! It's more clear (IMO) and more 'modern' :P